dali/pipeline/executor/executor.h

```diff
@@ -584,6 +594,13 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunGPU() {
     QueuePolicy::ReleaseIdxs(OpType::GPU, gpu_idxs);
     return;
   }
+
+  // short path for pure CPU pipeline
+  if (device_id_ < 0) {
```

Comment:
I'm thinking if some constant like `CPU_ONLY_DEVICE_ID = -1` should be introduced.

Reply:
On the other hand where to put it and where validate. I think there are many places where we are using this silently - like Tensor device and DeviceGuard. Hard to adjust all of them now. I would rather say that invalid GPU ID mean that it is CPU.